src/js/jsx/shared/Draggable.jsx

```diff
@@ -65,279 +65,295 @@ define(function (require, exports, module) {
-                } else {
-                    dndStore.updateDrag(this.props.zone, position);
+                    if (options.draggedTargets) {
+                        draggedTargets = options.draggedTargets;
+                    }
                 }
-            },
-
-            /**
-             * Handles finish of drag operation
-             * Removes drag event listeners from window
-             * Resets state
-             *
-             * @param {SyntheticEvent} event
-             */
-            _handleDragFinish: function (event) {
-                window.removeEventListener("mousemove", this._handleDragMove, true);
-                window.removeEventListener("mouseup", this._handleDragFinish, true);
-
-                this._mousedownPosition = null;
-
-                // Short circuit if not currently dragging
-                if (!this.state.dragging) {
+                
+                if (draggedTargets.isEmpty()) {
                     return;
                 }
 
-                // If the mouseup event is outside the window, there won't be an
-                // associated click event. In this case, remove the handler explicitly
-                // instead of waiting for (and suppressing) the following unrelated
-                // click event.
-                if (event.target === window.document.documentElement) {
-                    window.removeEventListener("click", this._handleDragClick, true);
-                } else {
-                    // The following click event sometimes never arrives, so just
-                    // remove the handler after a short timeout.
-                    window.setTimeout(function () {
-                        window.removeEventListener("click", this._handleDragClick, true);
-                    }.bind(this), 100);
-                }
+                // Suppress the following click event
+                window.addEventListener("click", this._handleDragClick, true);
 
-                if (this.props.onDragStop) {
-                    this.props.onDragStop();
-                }
-                
-                this.getFlux().store("draganddrop").stopDrag();
-            },
+                dndStore.startDrag(this.props.type, draggedTargets, position);
+            } else {
+                dndStore.updateDrag(position, element);
+            }
+        },
 
-            render: function () {
-                return (
-                    <Component
-                        {...this.props}
-                        isDragging={this.state.dragging}
-                        dragPosition={this.state.dragPosition}
-                        dragStyle={this.state.dragStyle}
-                        handleDragStart={this._handleDragStart} />
-                );
+        /**
+         * Handles finish of drag operation
+         * Removes drag event listeners from window
+         * Resets state
+         *
+         * @private
+         * @param {SyntheticEvent=} event
```

Comment:
This may just be a JSDoc error, but if this parameter is actually optional, it could throw below when dereferencing `target`.